landing-strip/core/src/main/java/com/novoda/landingstrip/TabsState.java

```diff
@@ -0,0 +1,189 @@
+package com.novoda.landingstrip;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.novoda.landingstrip.LandingStrip.TabSetterUpper;
+
+public class TabsState implements OnPagerAdapterChangedListener {
+
+    private static final int TAG_KEY_POSITION = R.id.ls__tag_key_position;
+
+    private final Attributes attributes;
+    private final LayoutInflater layoutInflater;
+    private final State state;
+    private final TabsContainer tabsContainer;
+    private final ViewGroup parent;
+    private final Scrollable scrollable;
+    private final IndicatorCoordinatesCalculator indicatorCoordinatesCalculator;
+
+    private ViewPager viewPager;
+    private PagerAdapterObserver pagerAdapterObserver;
+
+    public static TabsState newInstance(TabsContainer tabsContainer, Attributes attributes, ViewGroup parent, Scrollable scrollable) {
+        Context context = parent.getContext();
+        LayoutInflater layoutInflater = LayoutInflater.from(context);
+        State state = new State();
+        IndicatorCoordinatesCalculator indicatorCoordinatesCalculator = IndicatorCoordinatesCalculator.newInstance();
+
+        state.updateDelegateOnPageListener(new ViewPager.SimpleOnPageChangeListener());
+        state.updatePosition(0);
+        state.updatePositionOffset(0);
+        state.invalidateFastForwardPosition();
+
+        tabsContainer.attachTo(parent);
+        Builder builder = Builder.newInstance();
+        builder.attributes(attributes)
+                .indicatorCoordinatesCalculator(indicatorCoordinatesCalculator)
+                .layoutInflater(layoutInflater)
+                .parent(parent)
+                .state(state)
+                .scrollable(scrollable)
+                .tabsContainer(tabsContainer);
+        TabsState tabsState = builder.build();
+        PagerAdapterObserver observer = new PagerAdapterObserver(tabsState);
+        tabsState.setPagerAdapterObserver(observer);
+        return tabsState;
+    }
+
+    public Coordinates calculateIndicatorCoordinates() {
+        return indicatorCoordinatesCalculator.calculate(state.getPosition(), state.getPagePositionOffset(), tabsContainer);
+    }
+
+    private static class Builder {
```

Comment:
would prefer it to be suppressed with a `//TODO` or another refactoring to combine objects, such as the tab creation